server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/Decision.java

```diff
@@ -150,7 +152,7 @@ public boolean higherThan(Type other) {
     /**
      * Simple class representing a single decision
      */
-    public static class Single extends Decision {
+    public static class Single extends Decision implements ToXContentObject {
```

Comment:
To me it looks like the `Single` decisions are self-contained json, whereas the `Multi` decision is composed of several single decisions and just prints those in consecutive order. This seems to work because we alway seem to wrap them in an array (e.g. in MoveDecision#toXContent, NodeAllocationResult#toXContent). I think this is why the parent `Decision` is solely implementing `ToXContent` because the only thing guaranteed is that you can somehow deserialize it. The object hierarchy might be a bit off here but I think its out of scope for this PR to fix this.